o.n.core/src/org/netbeans/core/NbAuthenticator.java

```diff
@@ -71,7 +74,19 @@ private NbAuthenticator() {
 
     static void install() {
         if (Boolean.valueOf(NbBundle.getMessage(GuiRunLevel.class, "USE_Authentication"))) {
-            setDefault(new NbAuthenticator());
+            // Look for custom authenticator
+            Authenticator authenticator = Lookup.getDefault().lookup(Authenticator.class);
+            if (authenticator == null) {
+                authenticator = new NbAuthenticator();
+            }
+            if (authenticator.getClass().equals(NbAuthenticator.class)) {
```

Comment:
Everything else, including the log call doesn't seem necessary.

Reply:
After analyzing hundreds of messages files (NB log files) from my customers there's one thing I'm certain of: you can never have too much logging. :-)
Why do you think the logging calls are unnecessary?

Reply:
Because it's not usual to log which instance you are using from the `Lookup`. Since this is a static situation, you know at build-time which instance is going to be used, unless somebody yanks your whole module with the other implementation out.

Reply:
Why is it static?  The idea is that people can add their own Authenticators. Heck, I might even publish my own version as a plugin in the Plugin Portal and ask users to use my plugin if I believe I've come up with something brilliant which is better than the standard NbAuthenticator.  And that situation [already exists](https://bitbucket.org/phansson/netbeansnetworkauthenticator).

Reply:
I assumed only a Platform application would need that. I didn't think that people might want to install a 3rd party plugin in their existing IDE/Platform app.

Reply:
I can can think of a dozen reason why you might want to override the Authenticator on a *existing* application, i.e. use of Plugin. It is actually quite difficult to design an Authenticator which fits everyone's taste. For example imagine you are on a site where you have an alternative way that you can obtain credentials.

Reply:
I (and all my colleagues in my company) have serious trouble with during NB platform based applications such as the IDE itself, because we are sitting behind a corporate proxy that requires authentication. Please have a look at https://issues.apache.org/jira/browse/NETBEANS-58 for a detailed discussion. Therefore phansson solution should absolutely be included in NB otherwise users might have a very bad experience using NB IDE and will just get rid off in favor for alternatives. Bummer.

Reply:
Specifically, users on Windows, right?

Reply:
Yes